### DISCUSSION TITLE
Welcome to 2019 Discussions!

### DISCUSSION THREAD

**Discussion by bruce-armstrong** [OpenSourcePFCLibraries/2019#17] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mjl38** [comment]
Hi, my name is Miguel Leeuwe of Spanish-Dutch origins, living in the UK (at the moment) and I've been programming with powerbuilder since Jan. 1993.
My first question here is whether we should have the PBDOM objects included in the PFC. Personally I don't think we should, as everytime Appeon does a change, we'll have to update the pfc classes. As an alternative we can just use the PBD that comes with every version of powerbuilder or import the pbx into one of our own libraries. Maybe it's not that simple and there might be some dependencies?

regards

**Reply by brucearmstrong2** [reply]
When you "import" an PBNI extension all you do is create wrapper objects that in turn call the extension.  The only reason that those wrapper objects would change with changes to the extensions is if methods were added/deleted or their function signature changed.  I just did an import of the latest set of the extensions I have (build 2670) and there were absolutely no change to any of the proxy objects.

One thing I will do is remove the pbdom190 directory and it's contents.  That contains an old copy of the extension from when it was called pbdom190.  Appeon renamed that extension to just pbdom with the decoupling changes in they made in R3.  The current set of the wrapper objects already references the newer pbdom extension, so the pbdom190 wasn't being used, it was just extraneous.

**Reply by mjl38** [reply]
Hi Bruce, I think I already changed the objects (in pfcapsrv.pbl) previously to not include the "...190.pbx" anymore, so that would explain you didn't get any changes?
regards.

**Reply by mjl38** [reply]
https://github.com/OpenSourcePFCLibraries/2019/commit/33a4253f932523ac27670592d9e94371d3ff30b9

**Reply by brucearmstrong2** [reply]
Yes.  As a result we shouldn't have to touch those objects with any later major versions of PowerBuilder because they removed the version number reference.

FWIW, the main reason they added the ability to import the wrappers into a PBL was to support projects that are under source control.  I believe that having the wrapper objects imported in a PBL is the right approach for us.

**Reply by mjl38** [reply]
Yes, that's true. Okay, so let's keep them in the pfc classes then. My whole point was that someone has to remember to import the latest version all the time, but on the other hand it's nice to not have to import them yourself as a pfc user or find and attach the PBD in every project.
Thanks for a great job Bruce, much appreciated!

**Reply by brucearmstrong2** [reply]
>>My whole point was that someone has to remember to import the latest version all the time

My point is that we don't have to do that.  Not unless they add/remove or change the signature of functions.  With the runtime decoupling we don't even have to change it to support different major versions.

**Reply by mjl38** [reply]
Ok

**Comment by start2ar** [comment]
Hi Miguel, thats what i did in my project. i have removed the single pfc 
pbdom objects from the original pfc pbl and used the one provided inside 
PowerBuilder and i am not having any issues so far, but i am not sure if 
this is the right approach too


Il 09/04/2021 12.00, Miguel ha scritto:
>
> Hi, my name is Miguel Leeuwe and I've been programming with 
> powerbuilder since Jan. 1993.
> My first question here is whether we should have the PBDOM objects 
> included in the PFC. Personally I don't think we should, as everytime 
> Appeon does a change, we'll have to update the pfc classes. As an 
> alternative we can just use the PBD that comes with every version of 
> powerbuilder or import the pbx into one of our own libraries. Maybe 
> it's not that simple and there might be some dependencies?
>
> regards
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/OpenSourcePFCLibraries/2019/discussions/17#discussioncomment-588853>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AOG2AWOIIMCX2YMJLXU2AR3TH3F5DANCNFSM4XJVBZXA>.
>
-- 
<http://www.start2.it>
*Tel. 0575984910* 	
*Via F.lli Lumiere, 19* 	***@***.*** ***@***.***>
*52100 AREZZO (AR)* 	www.start2.it <http://www.start2.it>

**Reply by mjl38** [reply]
Same as we did here too,
regards
